scripts/swift/BuildLinks.swift

```diff
@@ -0,0 +1,85 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2019-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import Foundation
+import swsh
+import ArgumentParser
+
+struct BuildLinks: ParsableCommand {
+    static let configuration = CommandConfiguration(
+        abstract: "Create build links for QR codes in pull requests"
+    )
+
+    lazy var installPageMap =
+        Env.env["BITRISE_PUBLIC_INSTALL_PAGE_URL_MAP"] ??
+        Env.error("BITRISE_PUBLIC_INSTALL_PAGE_URL_MAP")
+
+    @Argument()
+    var branch: String
+
+    mutating func run() throws {
+        ExternalCommand.verbose = true
+        guard let prID = try Github.findAssociatedPullRequests(branch: branch).max() else {
+            print("can't find a pull request associated with branch \(branch)")
+            throw ExitCode.failure
+        }
+
+        let apps = installPageMap.components(separatedBy: "|")
+          .map { (x: String) -> [String] in x.components(separatedBy: "=>") }
+          .compactMap { (mapping: [String]) -> (file: String, url: String)? in
+              guard mapping.count == 2 else {
+                  return nil
+              }
+              let file = mapping[0].components(separatedBy: "/").last!
+              let url = mapping[1]
+              return (file: file, url: url)
+          }
+
+        var body = """
+          <!-- \(magicString) -->
+          """
+        for app in apps {
+            body += """
+                <details><summary>\(app.file)</summary>
+                \(app.url)
+                </details>
```

Comment:
The details, summary pattern means I need 2 clicks. Can we just make it a link?
`"<a href=\"\(app.url)\">\(app.file)</a>"`

Reply:
That was temporary testing code. It will be replaced by a QR image shortly